--- src/ppp.p.orig	Sat Dec 27 16:12:08 1997
+++ src/ppp.p	Sat Dec 27 16:12:58 1997
@@ -18,7 +18,7 @@
 int cifaddr  _P((int, int, int));
 int sifdefaultroute  _P((int, int));
 int cifdefaultroute  _P((int, int));
-int logwtmp _P((char *, char *, char *));
+extern void logwtmp _P((__const char *, __const char *, __const char *));
 int cifproxyarp  _P((int, u_long));
 int sifproxyarp  _P((int, u_long));
 int run_program _P((char *, char **, int));
--- src/ppp.c.orig	Sat Dec 27 16:57:04 1997
+++ src/ppp.c	Sat Dec 27 16:59:25 1997
@@ -858,11 +858,10 @@
         return 1;
 }
 
-int
+void
 logwtmp(line, name, host)
-	char *line, *name, *host;
+	__const char *line, *name, *host;
 {
-        return 1;
 }
 
 int
--- src/main.h.orig	Sat Dec 27 17:10:26 1997
+++ src/main.h	Sat Dec 27 17:10:56 1997
@@ -31,7 +31,7 @@
 
 extern char *exec_shell;
 extern u_int curtime;
-extern struct fd_set readfds, writefds, xfds;
+extern fd_set readfds, writefds, xfds;
 extern struct in_addr ctl_addr;
 extern struct in_addr special_addr;
 extern struct in_addr our_addr;
--- src/tcp_subr.c.orig	Sat Dec 27 18:30:34 1997
+++ src/tcp_subr.c	Sat Dec 27 18:43:51 1997
@@ -238,14 +238,13 @@
  * then send a RST to peer.
  */
 struct tcpcb *
-tcp_drop(tp, errno)
+tcp_drop(tp)
 	register struct tcpcb *tp;
-	int errno;
 {
 
 	DEBUG_CALL("tcp_drop");
 	DEBUG_ARG("tp = %lx", (long)tp);
-	DEBUG_ARG("errno = %d", errno);
+/*	DEBUG_ARG("errno = %d", errno); */
 	
 	if (TCPS_HAVERCVDSYN(tp->t_state)) {
 		tp->t_state = TCPS_CLOSED;
--- src/tcp_subr.p.orig	Sat Dec 27 18:56:37 1997
+++ src/tcp_subr.p	Sat Dec 27 18:57:05 1997
@@ -2,7 +2,7 @@
 void tcp_template _P((struct tcpcb *));
 void tcp_respond _P((struct tcpcb *, register struct tcpiphdr *, register struct mbuf *, tcp_seq, tcp_seq, int));
 struct tcpcb * tcp_newtcpcb _P((struct socket *));
-struct tcpcb * tcp_drop _P((register struct tcpcb *, int));
+struct tcpcb * tcp_drop _P((register struct tcpcb *));
 struct tcpcb * tcp_close _P((register struct tcpcb *));
 void tcp_drain _P((void));
 void tcp_sockclosed _P((struct tcpcb *));
--- src/tcp_input.c.orig	Sat Dec 27 18:54:26 1997
+++ src/tcp_input.c	Sat Dec 27 18:55:29 1997
@@ -725,7 +725,7 @@
 
 		if (tiflags & TH_RST) {
 			if (tiflags & TH_ACK)
-				tp = tcp_drop(tp,0); /* XXX Check t_softerror! */
+				tp = tcp_drop(tp); /* XXX Check t_softerror! */
 			goto drop;
 		}
 
@@ -967,7 +967,7 @@
 	 * error and we send an RST and drop the connection.
 	 */
 	if (tiflags & TH_SYN) {
-		tp = tcp_drop(tp,0);
+		tp = tcp_drop(tp);
 		goto dropwithreset;
 	}
 
--- src/tcp_timer.c.orig	Sat Dec 27 18:54:35 1997
+++ src/tcp_timer.c	Sat Dec 27 18:55:06 1997
@@ -196,7 +196,7 @@
 				 */
 				tp->t_rxtshift = TCP_MAXRXTSHIFT;
 				tcpstat.tcps_timeoutdrop++;
-				tp = tcp_drop(tp, tp->t_softerror);
+				tp = tcp_drop(tp);
 				/* tp->t_softerror : ETIMEDOUT); */ /* XXX */
 				return (tp); /* XXX */
 			}
@@ -321,7 +321,7 @@
 
 	dropit:
 		tcpstat.tcps_keepdrops++;
-		tp = tcp_drop(tp, 0); /* ETIMEDOUT); */
+		tp = tcp_drop(tp);
 		break;
 	}
 
